### PR TITLE
CODING_21175_UPDATE_USER_INFO

### PR DESCRIPTION
**Pull request by pham-tam-mulodo**
* Code function
    - [x] put_update() [Controller]
    - [x] update_user() [Auth]
* Code unit test
    - [x] put_update()

### CODE REVIEW

File: src/fuel/app/tests/controller/v1/users.php [hunk 1]
```diff
@@ -6,6 +9,17 @@
...
  */
 class Test_Controller_V1_Users extends TestCase {
+    private $_link = 'http://miniblog.tam/v1/';
+
+    protected function setUp() {
+
+    }
+
+    protected function tearDown()
+    {
```
Comment (truong-vu-mulodo) [inline src/fuel/app/tests/controller/v1/users.php]: Please check coding standard.

File: src/fuel/packages/auth/classes/auth/login/simpleauth.php [hunk 2]
```diff
@@ -20,596 +20,556 @@
...
+            }
+
+            // return true when login was verified, and either the hash matches or multiple logins are allowed
+            if ($this->user and (\Config::get('simpleauth.multiple_logins', false) or $this->user['login_hash'] === $login_hash)) {
+                return true;
+            }
+        }
+
+        // not logged in, do we have remember-me active and a stored user_id?
+        elseif (static::$remember_me and $user_id = static::$remember_me->get('user_id', null)) {
```
Comment (truong-vu-mulodo) [inline src/fuel/packages/auth/classes/auth/login/simpleauth.php]: Please check coding standard.

File: src/fuel/packages/auth/classes/auth/login/simpleauth.php [hunk 3]
```diff
@@ -20,596 +20,556 @@
...
+            ->from(\Config::get('simpleauth.table_name'))
+            ->execute(\Config::get('simpleauth.db_connection'));
+
+        if ($same_users->count() > 0) {
+            if (in_array(strtolower($email), array_map('strtolower', $same_users->current()))) {
+                $res['code'] = ERROR_EMAIL_EXIST;
+                $res['message'] = MSG_EMAIL_EXIST;
+                return $res;
+            }
+            else {
```
Comment (truong-vu-mulodo) [inline src/fuel/packages/auth/classes/auth/login/simpleauth.php]: Please check coding standard.

File: src/fuel/app/tests/controller/v1/users.php (hunk 1 above)
Comment (pham-tam-mulodo) [inline src/fuel/app/tests/controller/v1/users.php]: sorry, i test Mock, and forget remove.

File: src/fuel/packages/auth/classes/auth/login/simpleauth.php [hunk 4]
```diff
@@ -20,596 +20,556 @@
...
+            'mobile'            => (string) $profile_fields['mobile'],
+            'last_login'        => 0,
+            'login_hash'        => '',
+            'created_at'        => (string) date('Y-m-d H:i:s',time()),
+            'updated_at'        => (string) date('Y-m-d H:i:s',time())
+        );
+        $result = \DB::insert(\Config::get('simpleauth.table_name'))
+            ->set($user)
+            ->execute(\Config::get('simpleauth.db_connection'));
+        if ($result[1] > 0)
```
Comment (truong-vu-mulodo) [inline src/fuel/packages/auth/classes/auth/login/simpleauth.php]: Please check coding standard.

File: src/fuel/packages/auth/classes/auth/login/simpleauth.php [hunk 5]
```diff
@@ -20,596 +20,556 @@
...
+            $res['code'] = STATUS_OK;
+        else {
```
Comment (truong-vu-mulodo) [inline src/fuel/packages/auth/classes/auth/login/simpleauth.php]: Please check coding standard.

File: src/fuel/packages/auth/classes/auth/login/simpleauth.php [hunk 6]
```diff
@@ -20,596 +20,556 @@
...
+                ->where('username', '=', $username)
+                ->from(\Config::get('simpleauth.table_name'))
+                ->execute(\Config::get('simpleauth.db_connection'))->current();
+        }
+
+        if ($affected_rows > 0) {
+            $res['code'] = STATUS_OK;
+            $res['data'] = $this->user;
+        }
+        else {
```
Comment (truong-vu-mulodo) [inline src/fuel/packages/auth/classes/auth/login/simpleauth.php]: Please check coding standard.

File: src/fuel/packages/auth/classes/auth/login/simpleauth.php [hunk 7]
```diff
@@ -20,596 +20,556 @@
...
+     * @param  string  name of the user field to return
+     * @param  mixed  value to return if the field requested does not exist
+     *
+     * @return  mixed
+     */
+    public function get($field, $default = null) {
+        if (isset($this->user[$field])) {
+            return $this->user[$field];
+        }
+        elseif (isset($this->user['profile_fields'])) {
```
Comment (truong-vu-mulodo) [inline src/fuel/packages/auth/classes/auth/login/simpleauth.php]: Please check coding standard.

File: src/fuel/packages/auth/classes/auth/login/simpleauth.php [hunk 8]
```diff
@@ -20,596 +20,556 @@
...
+     */
+    public function get_profile_fields($field = null, $default = null) {
+        if (empty($this->user)) {
+            return false;
+        }
+
+        if (isset($this->user['profile_fields'])) {
+            is_array($this->user['profile_fields']) or $this->user['profile_fields'] = (@unserialize($this->user['profile_fields']) ?: array());
+        }
+        else {
```
Comment (truong-vu-mulodo) [inline src/fuel/packages/auth/classes/auth/login/simpleauth.php]: Please check coding standard.

**Comment by truong-vu-mulodo** [comment]
@pham-tam-mulodo: Just go ahead after done my comments.

**Pull request merged**